### PR TITLE
IGNITE-14305 Fix snapshot check command for indexes

### PR DESCRIPTION
**Pull request by Mmuzaf**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import org.junit.Test;
+
+import static java.util.Collections.singletonList;
+import static org.apache.ignite.testframework.GridTestUtils.assertContains;
+
+/**
+ * Cluster-wide snapshot test check command with indexes.
+ */
+public class IgniteClusterSnapshotCheckWithIndexesTest extends AbstractSnapshotSelfTest {
+    private final CacheConfiguration<Integer, Account> indexedCcfg =
```
Comment (xtern) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java:37]: javadoc is missing

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        assertTrue("Exceptions: " + b, F.isEmpty(res.exceptions()));
+        assertTrue(F.isEmpty(res.exceptions()));
+    }
+
+    /** @throws Exception If fails. */
+    @Test
+    public void testClusterSnapshotCheckWithIndexes() throws Exception {
+        IgniteEx ignite = startGridsWithCache(3, CACHE_KEYS_RANGE, key -> new Account(key, key), indexedCcfg);
+
+        ignite.snapshot().createSnapshot(SNAPSHOT_NAME).get();
```
Comment (xtern) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java:62]: I suggest adding a short timeout for all places where we are waiting for something

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotMetadata.java [hunk 3]
```diff
@@ -137,7 +137,7 @@ public int pageSize() {
     }
 
     /**
-     * @return The list of cache groups ids which were included into snapshot.
+     * @return The list of cache groups ids which were included into the snapshot globally.
```
Comment (xtern) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotMetadata.java:140]: "cache groups ids" can be changed to "cache group IDs"
it's up to you, but I found the expression "cache group ids" only in snapshots

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java (hunk 1 above)
Comment (Mmuzaf) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java:37]: Fixed.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java (hunk 2 above)
Comment (Mmuzaf) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java:62]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotMetadata.java (hunk 3 above)
Comment (Mmuzaf) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotMetadata.java]: Fixed.

**Comment by Mmuzaf** [comment]
@xtern I've fixed all your comments, please, take a look.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+        StringBuilder b = new StringBuilder();
+        res.print(b::append, true);
+
+        assertTrue("Exceptions: " + b, F.isEmpty(res.exceptions()));
+        assertContains(log, b.toString(), "The check procedure has finished, no conflicts have been found.");
+    }
+
+    /** Node filter to run cache on single node. */
+    private static class SelfNodeFilter implements IgnitePredicate<ClusterNode> {
```
Comment (xtern) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java:100]: SerialVersionUID is missing

**Review by xtern — APPROVED** [review]

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java (hunk 4 above)
Comment (Mmuzaf) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotCheckWithIndexesTest.java:100]: Fixed.

**Pull request merged**